cpp/src/table/row_operators.cu

```diff
@@ -373,51 +391,391 @@ namespace row {
+                                    new_column_order_lhs,
+                                    new_null_precedence_lhs,
+                                    has_ranked_children_lhs,
+                                    stream),
+          create_preprocessed_table(transformed_rhs_opt.value(),
+                                    std::move(verticalized_col_depths_rhs),
+                                    std::move(transformed_columns_rhs),
+                                    new_column_order_lhs,
+                                    new_null_precedence_lhs,
+                                    has_ranked_children_rhs,
+                                    stream)};
+}
+
+preprocessed_table::preprocessed_table(
+  table_device_view_owner&& table,
+  rmm::device_uvector<order>&& column_order,
+  rmm::device_uvector<null_order>&& null_precedence,
+  rmm::device_uvector<size_type>&& depths,
+  std::vector<detail::dremel_data>&& dremel_data,
+  rmm::device_uvector<detail::dremel_device_view>&& dremel_device_views,
+  std::vector<std::unique_ptr<column>>&& transformed_columns,
+  bool has_ranked_children)
+  : _t(std::move(table)),
+    _column_order(std::move(column_order)),
+    _null_precedence(std::move(null_precedence)),
+    _depths(std::move(depths)),
+    _dremel_data(std::move(dremel_data)),
+    _dremel_device_views(std::move(dremel_device_views)),
+    _transformed_columns(std::move(transformed_columns)),
+    _has_ranked_children(has_ranked_children)
+{
+}
+
+preprocessed_table::preprocessed_table(table_device_view_owner&& table,
+                                       rmm::device_uvector<order>&& column_order,
+                                       rmm::device_uvector<null_order>&& null_precedence,
+                                       rmm::device_uvector<size_type>&& depths,
+                                       std::vector<std::unique_ptr<column>>&& transformed_columns,
+                                       bool has_ranked_children)
+  : _t(std::move(table)),
+    _column_order(std::move(column_order)),
+    _null_precedence(std::move(null_precedence)),
+    _depths(std::move(depths)),
+    _dremel_data{},
+    _dremel_device_views{},
+    _transformed_columns(std::move(transformed_columns)),
+    _has_ranked_children(has_ranked_children)
+{
+}
+
 two_table_comparator::two_table_comparator(table_view const& left,
                                            table_view const& right,
                                            host_span<order const> column_order,
                                            host_span<null_order const> null_precedence,
                                            rmm::cuda_stream_view stream)
-  : d_left_table{preprocessed_table::create(left, column_order, null_precedence, stream)},
-    d_right_table{preprocessed_table::create(right, column_order, null_precedence, stream)}
 {
-  check_shape_compatibility(left, right);
+  std::tie(d_left_table, d_right_table) =
```

Comment:
Can you use a structured binding here instead of `std::tie`?

Reply:
Unfortunately, structure binding can't be used to bind the output to existing variables.